benchmark_test.go

```diff
@@ -0,0 +1,64 @@
+package wparams_test
+
+import (
+	"context"
+	"testing"
+
+	wparams "github.com/palantir/witchcraft-go-params"
+)
+
+func BenchmarkContextWithSafeParam(b *testing.B) {
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		ctx := context.Background()
+		ctx = wparams.ContextWithSafeParam(ctx, "val1", 1)
+		ctx = wparams.ContextWithSafeParam(ctx, "val2", 2)
+		ctx = wparams.ContextWithSafeParam(ctx, "val3", 3)
+		_, _ = wparams.SafeAndUnsafeParamsFromContext(ctx)
```

Comment:
Does assigning to `_` avoid the compiler optimizations? I used to always use the approach of assigning to pkg level variables, but this would be much cleaner.

Reply:
Yeah it seems to do all the work, this is the pattern I've used in other benchmarks